code/api_crawler.py

```diff
@@ -54,21 +54,62 @@ def getOperationId(uri):
             return operationId
 
 
+def getValuesByPath(propertyName, propertyList, valueSource):
+    if debug==True:
+        print "getValuesByPath:", propertyName, propertyList
+
+    value = valueSource.get(propertyName)
+
+    if debug==True:
+        print "~~~> value:", value
+
+    if propertyList.index(propertyName)<(len(propertyList)-1):
+        nextProperty = propertyList[propertyList.index(propertyName)+1]
+    else:
+        nextProperty = None
+
+    if value is None:
+        return []
+    elif type(value)==type([]):
+        valueList = []
+        for valueItem in value:
+            if nextProperty is not None:
+                valueList.extend(getValuesByPath(nextProperty, propertyList[propertyList.index(nextProperty):], valueItem))
+            else:
+                valueList.append(valueItem)
+
+        return valueList
+    else:
+        if nextProperty is not None:
+            return getValuesByPath(nextProperty, propertyList[propertyList.index(nextProperty):], value)
+        else:
+            return value
+
+
+
 def expandTemplatedUri(uri, response):
     if debug==True:
         print "expandTemplatedUri:", uri
```

Comment:
Om het script werkend te krijgen in windows, moet ik elke print commando aanpassen door haakjes om de parameters te zetten.
`print "expandTemplatedUri:", uri` -> `print("expandTemplatedUri:", uri)`

Dit geldt uiteraard ook voor de andere scripts.

Reply:
ik draai python 2.7, jij blijkbaar python 3. Vandaar de noodzaak voor de haakjes. Ik zal ze toevoegen